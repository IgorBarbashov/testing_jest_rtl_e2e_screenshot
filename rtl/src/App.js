import React from 'react';
import {BrowserRouter, Link, Route, Routes} from "react-router-dom";
import AboutPage from "./pages/AboutPage";
import MainPage from "./pages/MainPage";

const App = () => {
    return (
        <BrowserRouter>
            <Link to='/' data-testid='main-link'>main</Link>
            {' '}
            <Link to='/about' data-testid='about-link'>about</Link>
            <Routes>
                <Route path='/' element={<MainPage />} />
                <Route path='/about' element={<AboutPage />} />
            </Routes>
        </BrowserRouter>
    );
};

export default App;

// import React, {useState, useEffect} from 'react';
//
// const App = () => {
//     const [data, setData] = useState(null);
//     const [toggle, setToggle] = useState(false);
//     const [value, setValue] = useState('');
//
//     const onClick = () => setToggle(prev => !prev);
//     const onChange = (e) => setValue(e.target.value);
//
//     useEffect(() => {
//         const stid = setTimeout(() => {
//             setData({});
//         }, 100);
//         return () => clearTimeout(stid);
//     }, []);
//
//     return (
//         <div>
//             {<h2 data-testid="input-header">{value}</h2>}
//             {data && <div>data</div>}
//             {toggle && <div data-testid="toggle-elem">toggle</div>}
//             <h1>Hello world</h1>
//             <button data-testid="toggle-btn" onClick={onClick}>click me</button>
//             <input
//                 style={{color: 'red'}}
//                 type="text"
//                 placeholder="enter value"
//                 onChange={onChange}
//             />
//         </div>
//     );
// };
//
// export default App;