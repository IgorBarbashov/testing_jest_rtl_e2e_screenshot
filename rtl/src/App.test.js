import {render, screen} from '@testing-library/react';
import App from './App';

describe('App component', () => {
    test('should render fields', () => {
        render(<App />);
        const helloWorldElement = screen.getByText(/hello world/i);
        const buttonElement = screen.getByRole('button');
        const inputElement = screen.getByPlaceholderText(/enter value/i);
        expect(helloWorldElement).toBeInTheDocument();
        expect(buttonElement).toBeInTheDocument();
        expect(inputElement).toBeInTheDocument();

        screen.debug();
    });

    test('should correctly render field input', () => {
        render(<App />);
        const inputElement = screen.getByPlaceholderText(/enter value/i);
        expect(inputElement).toMatchSnapshot();
    });

    test('negative test', () => {
        render(<App />);
        const inputElement = screen.queryByText(/try to find me/i);
        expect(inputElement).not.toBeInTheDocument();
        expect(inputElement).toBeNull();
    });
});