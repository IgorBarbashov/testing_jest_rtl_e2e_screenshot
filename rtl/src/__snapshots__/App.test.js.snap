// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component negative test 1`] = `
<input
  placeholder="enter value"
  type="text"
/>
`;

exports[`App component negftive test 1`] = `
<input
  placeholder="enter value"
  type="text"
/>
`;

exports[`App component should correctly render field input 1`] = `
<input
  placeholder="enter value"
  type="text"
/>
`;
